### PR TITLE
refactor: remove calls to DbInterface._rawExec

### PR DESCRIPTION
**Pull request by thelindat**
**Pull Request Description**

Start replacing the calls to the deprecated DbInterface._rawExec function and ensure proper typecasting.
- Utilise pool.execute over pool.query (may need to adjust some queries)
- Ensure proper return values/types

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

**Comment by thelindat** [comment]
I'm testing all the queries at the moment, but I'm likely to miss stuff if @RockySouthpaw wants to double-check things.

**Comment by TasoOneAsia** [comment]
Does latest node not require the scheduling tick to speed up IO callback dispatching?

**Comment by thelindat** [comment]
> Does latest node not require the scheduling tick to speed up IO callback dispatching?

I don't believe it's necessary for internal use, rather when doing exports and having to wait for the response. Even without it in oxmysql it's fast until you throw a wait in your function.

File: resources/server/db/db_wrapper.ts [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+        this.logger.info(
+          `${query} (${values}) execution time: ${process.hrtime(startTime)[1] / 1000000}ms`,
+        );
+        return result;
+      }
 
-      ScheduleResourceTick(RESOURCE_NAME);
-      return await pool.query(query, values);
+      const result = await conn.execute(query, values);
+      conn.release();
```
Comment (TasoOneAsia) [inline resources/server/db/db_wrapper.ts:24]: Whats the reason for managing the connection release manually? Isn't this wrapped internally within `mysql` query wrapper?

File: resources/server/db/db_wrapper.ts [hunk 2]
```diff
@@ -38,8 +44,8 @@ class _DbInterface {
    * @param values Variable definition
    **/
   public async exec(query: string, values?: unknown) {
-    const [res] = await this._internalQuery(query, values);
-    return (<ResultSetHeader>res).affectedRows;
+    const [res] = (await this._internalQuery(query, values)) as ResultSetHeader[];
+    return res.affectedRows;
   }
```
Comment (TasoOneAsia) [inline resources/server/db/db_wrapper.ts:49]: The exec methods return types here could probably be done a little bit better here, using passed generics or something but tbh this works too, just explicit type assertion can feel a little *hacky* when compared to a generic system imo, need to think on long term implementation details but as a little hack this is fine.

File: resources/server/marketplace/marketplace.db.ts [hunk 3]
```diff
@@ -23,44 +22,42 @@ export class _MarketplaceDB {
...
     const query = `DELETE FROM npwd_marketplace_listings WHERE identifier = ?`;
-    await DbInterface._rawExec(query, [identifier]);
+    await DbInterface.exec(query, [identifier]);
   }
 
   async getListing(listingId: number): Promise<MarketplaceListing> {
     const query = `SELECT * FROM npwd_marketplace_listings WHERE id = ?`;
-    const [results] = await DbInterface._rawExec(query, [listingId]);
+    const results = await DbInterface.fetch(query, [listingId]);
     const listings = <MarketplaceListing[]>results;
```
Comment (TasoOneAsia) [inline resources/server/marketplace/marketplace.db.ts:49]: As per last review comment, this method would benefit from a passable generic type here so we don't need to assert the result.

**Review by TasoOneAsia — COMMENTED** [review]
Just a couple things to think about, do you have the updated IO cb query metrics when not manually scheduling another resource tick now?

**Comment by thelindat** [comment]
Going to redo this stuff in the future once changes have settled down.

**Pull request closed without merging**